src/player.cpp

```diff
@@ -2283,9 +2280,7 @@ stats player::get_stats() const
 
 void player::mod_stat( const std::string &stat, int modifier )
 {
-    if( stat == "hunger" ) {
-        hunger += modifier;
```

Comment:
You forgot to add this back in `Character::mod_stat`, currently the "hunger" stat is not handled anywhere.